File: webstore/webstore-java/src/main/java/com/renan/webstore/business/impl/UserServiceImpl.java
package com.renan.webstore.business.impl;

import com.renan.webstore.business.AppErrorType;
import com.renan.webstore.business.BusinessException;
import com.renan.webstore.business.UserService;
import com.renan.webstore.business.ValidatorService;
import com.renan.webstore.controller.form.user.ChangePasswordForm;
import com.renan.webstore.model.AppUser;
import com.renan.webstore.persistence.repository.UserRepository;
import com.renan.webstore.util.MessageUtil;
import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
@RequiredArgsConstructor
public class UserServiceImpl implements UserService {
    
    private final UserRepository repository;
    private final PasswordEncoder passwordEncoder;
    private final ValidatorService validatorService;
    @Override
    public void updateUserPassword(ChangePasswordForm form, Integer userID) throws BusinessException {
        validateChangePasswordInput(form, userID);
        Optional<AppUser> userOpt = repository.findById(userID);
        if (userOpt.isEmpty()) throw new BusinessException(
                MessageUtil.getErrorMessageUserNotFound(),
                BusinessException.BUSINESS_MESSAGE,
                AppErrorType.INVALID_INPUT
        );
        AppUser user = userOpt.get();
        user.setPassword(passwordEncoder.encode(form.getNewPassword().trim()));
        repository.save(user);
    }
    
    private void validateChangePasswordInput(ChangePasswordForm form, Integer userID) throws BusinessException {
        if (userID == null || userID <= 0 || form == null || !validatorService.isValidPassword(form.getNewPassword()) ) {
            throw new BusinessException(
                    MessageUtil.getErrorMessageInputValues(),
                    BusinessException.BUSINESS_MESSAGE,
                    AppErrorType.INVALID_INPUT
            );
        }
    }
    
}

File: webstore/webstore-kotlin/src/main/kotlin/com/renan/webstore/business/impl/UserServiceImpl.kt
package com.renan.webstore.business.impl

import com.renan.webstore.business.AppErrorType
import com.renan.webstore.business.BusinessException
import com.renan.webstore.business.UserService
import com.renan.webstore.business.validator.Validator
import com.renan.webstore.controller.user.form.UserNameForm
import com.renan.webstore.controller.user.form.UserPasswordForm
import com.renan.webstore.persistence.dao.UserDAO
import com.renan.webstore.util.getErrorMessageInputValues
import com.renan.webstore.util.getErrorMessageUserNotFound
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service

@Service
class UserServiceImpl(
    val userDAO: UserDAO,
    val validator: Validator,
    val passwordEncoder: PasswordEncoder
) : UserService {

    override fun updateUserPassword(form: UserPasswordForm, userId: Int) {
        if (!validator.isValidPassword(form.newPassword)) throw BusinessException(
            message = getErrorMessageInputValues(),
            businessMessage = true,
            errorType = AppErrorType.INVALID_INPUT
        )
        val optUser = userDAO.findById(userId)
        if (optUser.isEmpty) throw BusinessException(
            message = getErrorMessageUserNotFound(),
            businessMessage = true,
            errorType = AppErrorType.INVALID_INPUT
        )
        val user = optUser.get()
        user.userPassword = passwordEncoder.encode(form.newPassword)
        userDAO.saveUser(user)
    }

    override fun updateUserName(form: UserNameForm, userId: Int) {
        if (form.name.isBlank()) throw BusinessException(
            message = getErrorMessageInputValues(),
            businessMessage = true,
            errorType = AppErrorType.INVALID_INPUT
        )
        val optUser = userDAO.findById(userId)
        if (optUser.isEmpty) throw BusinessException(
            message = getErrorMessageUserNotFound(),
            businessMessage = true,
            errorType = AppErrorType.INVALID_INPUT
        )
        val user = optUser.get()
        user.firstName = form.name.trim()
        user.lastName = form.lastName?.trim()
        userDAO.saveUser(user)
    }

}